python/lsst/daf/butler/registry/interfaces/_opaque.py

```diff
@@ -48,14 +51,17 @@ def __init__(self, name: str):
         self.name = name
 
     @abstractmethod
-    def insert(self, *data: dict) -> None:
+    def insert(self, *data: dict, transaction: DatastoreTransaction | None = None) -> None:
```

Comment:
In some sense we need a `Transaction` abstract base class for this rather than `DatastoreTransaction`. The problem with a lot of our storage classes is that they are assuming they are not really responsible for transaction handling and that the database will deal with it all. Whenever something turns up that doesn't have a SQL database backing it we run into a problem. Datastore has a transaction object but it has no way to pass that on to registry without having an API for it. Where else should I put the undo if Datastore doesn't know whether the records table is going to sort itself out because of a database transaction or needs help to manually undo?

We had to make the same sort of change in `DatastoreRegistryBridge` in #733 but that was at least a datastore bridge getting a datastore transaction.

Reply:
After discussion I added more detailed docs to the insert method explaining the transaction.